packages/ethereum-contracts/contracts/superfluid/SuperTokenFactory.sol

```diff
@@ -18,26 +19,44 @@ import { SuperToken } from "../superfluid/SuperToken.sol";
 
 import { FullUpgradableSuperTokenProxy } from "./FullUpgradableSuperTokenProxy.sol";
 
-import { Address } from "@openzeppelin/contracts/utils/Address.sol";
-import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
-
-
 abstract contract SuperTokenFactoryBase is
     UUPSProxiable,
     ISuperTokenFactory
 {
 
+    /* WARNING: NEVER RE-ORDER VARIABLES! Including the base contracts.
+        Always double-check that new
+        variables are added APPEND-ONLY. Re-ordering variables can
+        permanently BREAK the deployed proxy contract. */
+
     ISuperfluid immutable internal _host;
 
     ISuperToken internal _superTokenLogic;
 
+    /// @notice A mapping from underlying token addresses to canonical wrapper super token addresses
+    /// @dev Reasoning: (1) provide backwards compatibility for existing listed wrapper super tokens
+    /// @dev (2) prevent address retrieval issues if we ever choose to modify the bytecode of the UUPSProxy contract
+    /// @dev NOTE: address(0) key points to the NativeAssetSuperToken on the network.
+    mapping(address => address) internal _canonicalWrapperSuperTokens;
+    
+    /// NOTE: Whenever modifying the storage layout here it is important to update the validateStorageLayout
+    /// function in its respective mock contract to ensure that it doesn't break anything or lead to unexpected
+    /// behaviors/layout when upgrading
+
     constructor(
         ISuperfluid host
     ) {
         _host = host;
     }
 
-    /// @dev ISuperTokenFactory.getHost implementation
+    modifier onlyGovernanceOwner() {
+        SuperfluidGovernanceII gov = SuperfluidGovernanceII(address(_host.getGovernance()));
+        address owner = gov.owner();
+        if (msg.sender != owner) revert SuperfluidGovernanceII.SF_GOV_II_ONLY_OWNER();
+        _;
+    }
```

Comment:
We shouldn't make this dependent on `SuperfluidGovernanceII`, is not used e.g. on testnets and may change in the future.
We have a problem here: neither `ISuperfluidGovernance` nor `SuperfluidGovernanceBase` include `Ownable`. But all implementations we have in use do. @hellwolf Shall we establish that going forward all gov implementations have to implement `Ownable` and use `Ownable` here?

Reply:
agreed